Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskRepository.kt

```diff
@@ -51,37 +64,76 @@ class PresenceTracingRiskRepository @Inject constructor(
             presenceTracingRiskCalculator.calculateAggregatedRiskPerDay(it)
         }
 
-    suspend fun replaceAllMatches(list: List<CheckInWarningOverlap>) {
-        deleteAllMatches()
-        addAll(list)
+    internal suspend fun reportSuccessfulCalculation(list: List<CheckInWarningOverlap>) {
+        traceTimeIntervalMatchDao.insert(list.map { it.toEntity() })
+        val last14days = normalizedTime.first().filter { it.localDateUtc.isAfter(fifteenDaysAgo.toLocalDateUtc()) }
+        val risk = presenceTracingRiskCalculator.calculateTotalRisk(last14days)
+        add(
+            PtRiskLevelResult(
+                timeStamper.nowUTC,
+                risk
+            )
+        )
     }
 
-    private suspend fun addAll(list: List<CheckInWarningOverlap>) {
-        traceTimeIntervalMatchDao.insert(list.map { it.toEntity() })
+    internal suspend fun deleteStaleData() {
+        traceTimeIntervalMatchDao.deleteOlderThan(fifteenDaysAgo.millis)
+        riskLevelResultDao.deleteOlderThan(fifteenDaysAgo.millis)
     }
 
-    suspend fun deleteStaleMatches() {
-        val endTime = timeStamper.nowUTC.minus(Days.days(15).toStandardDuration())
-        traceTimeIntervalMatchDao.deleteOlderThan(endTime.millis)
+    internal suspend fun markPackageProcessed(warningPackageId: String) {
+        // TODO
```

Comment:
when/where this will be implemented?

Reply:
When @d4rken has finished https://github.com/corona-warn-app/cwa-app-android/pull/2707 due to a missing db field